envoy/http/async_client.h

```diff
@@ -318,6 +364,10 @@ class AsyncClient {
       sampled_ = sampled;
       return *this;
     }
+    RequestOptions& setEndStream(bool end_stream) {
+      end_stream_immediately_ = end_stream;
```

Comment:
it's unintuitive to me that we have to latch end stream immediately rather than ending the stream immediately.  can we make more clear what'll happen and when?  Is this really just a boolean to indicate if the stream is headers only?

Reply:
Fixed -- removed the end stream field and documented that for a headers only request, the `send` method should be used instead of `startRequest`.